File: utils/latin_square.py
LATIN_SQUARE = [
    {
        "conditions": ["A", "B", "C", "D"],
        "scenarios": ["temporal", "relational", "face", "grief"],
    },
    {
        "conditions": ["B", "C", "D", "A"],
        "scenarios": ["relational", "face", "grief", "temporal"],
    },
    {
        "conditions": ["C", "D", "A", "B"],
        "scenarios": ["face", "grief", "temporal", "relational"],
    },
    {
        "conditions": ["D", "A", "B", "C"],
        "scenarios": ["grief", "temporal", "relational", "face"],
    },
]


def get_assignment(row: int) -> dict:
    return LATIN_SQUARE[row % 4]

File: web/src/lib/latin-square.ts
import type { Condition, ScenarioType } from "./types";

const LATIN_SQUARE: { conditions: Condition[]; scenarios: ScenarioType[] }[] = [
  {
    conditions: ["A", "B", "C", "D"],
    scenarios: ["scenario_1", "scenario_2", "scenario_3", "grief"],
  },
  {
    conditions: ["B", "C", "D", "A"],
    scenarios: ["scenario_2", "scenario_3", "grief", "scenario_1"],
  },
  {
    conditions: ["C", "D", "A", "B"],
    scenarios: ["scenario_3", "grief", "scenario_1", "scenario_2"],
  },
  {
    conditions: ["D", "A", "B", "C"],
    scenarios: ["grief", "scenario_1", "scenario_2", "scenario_3"],
  },
];

export function getAssignment(row: number) {
  return LATIN_SQUARE[row % 4];
}
